### PR TITLE
Feature/bib 567 unique

### PR DESCRIPTION
**Pull request by tuj**
https://jira.itkdev.dk/browse/BIB-567

### CODE REVIEW

File: src/Controller/BoxConfigurationController.php [hunk 1]
```diff
@@ -17,18 +17,18 @@
 class BoxConfigurationController extends AbstractController
 {
     /**
-     * @Route("/box/configuration/{id}", name="box_configuration")
+     * @Route("/box/configuration/{uniqueId}", name="box_configuration")
      *
-     * @param $id
-     *   Box configuration ID
+     * @param $uniqueId
```
Comment (cableman) [inline src/Controller/BoxConfigurationController.php:22]: Missing type

File: src/Controller/BoxConfigurationController.php [hunk 2]
```diff
@@ -17,18 +17,18 @@
...
+     *   Box configuration unique id
      * @param BoxConfigurationRepository $boxConfigurationRepository
      *   Box configuration repository
      *
      * @return JsonResponse
      */
-    public function index($id, BoxConfigurationRepository $boxConfigurationRepository)
+    public function index($uniqueId, BoxConfigurationRepository $boxConfigurationRepository)
```
Comment (cableman) [inline src/Controller/BoxConfigurationController.php:29]: Missing type on `$uniqueId`

File: src/EventSubscriber/BoxConfigurationEventListener.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                $bytes = random_bytes(5);
+                $newUniqueId = bin2hex($bytes);
+
+                // Test for uniqueness.
+                $entitiesFound = $this->boxConfigurationRepository->findBy(['uniqueId' => $newUniqueId]);
+                if (0 === \count($entitiesFound)) {
+                    $uniqueId = $newUniqueId;
+                }
+            } catch (\Exception $e) {
+            }
```
Comment (cableman) [inline src/EventSubscriber/BoxConfigurationEventListener.php:69]: You could lock at hash function a strip it down like git do with it's SHA1

**Review by cableman — CHANGES_REQUESTED** [review]

File: src/Entity/BoxConfiguration.php [hunk 4]
```diff
@@ -116,6 +116,11 @@ class BoxConfiguration
      */
     private $defaultLanguageCode = 'EN';
 
+    /**
+     * @ORM\Column(type="string", length=255)
```
Comment (cableman) [inline src/Entity/BoxConfiguration.php:120]: If the length is only 10 chars no need to make space for 255 chars!

**Review by cableman — CHANGES_REQUESTED** [review]

File: src/Entity/BoxConfiguration.php (hunk 4 above)
Comment (tuj) [reply src/Entity/BoxConfiguration.php:120]: It leaves space for other choices of lengths.

**Review by cableman — APPROVED** [review]

**Pull request merged**